src/FilenameParsing/NaturalFileIDHelper.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace SilverStripe\Assets\FilenameParsing;
+
+use SilverStripe\Core\Injector\Injectable;
+
+/**
+ * Parsed Natural path URLs. Natural path are the same hashless path that appear in the CMS.
+ *
+ * Natural paths are used by the public adapter from SilverStripe 4.4 and on the protected adapter when
```

Comment:
The difference between legacy_filenames, and this, is that protected files don't need to be moved during update. Even if the public files are the same, protected files will be "broken" until manually moved. That's what `legacy_filenames` is intended to do.

What you should do is change `legacy_filenames` so that it forces the same public resolution strategy, but for private files, and that will prevent regressions. :)

Reply:
We had a discussion about what to do with `legacy_filenames`. We decided to ignore `legacy_filenames` going forward. Everyone will use natural path for public and hash path for protected files.

Because both path type will be resolved on both store now, the files can stay where they are without issues.

If someone really wanted to keep natural path on the protected store, they could update the protected resolution strategy to default to NaturalFileIDHelper.